lib/Parse/Parser.cpp

```diff
@@ -1016,20 +1016,18 @@ Parser::parseList(tok RightK, SourceLoc LeftLoc, SourceLoc &RightLoc,
       break;
     }
 
-    // If the next token is at the beginning of a new line, could be an argument
-    // label, is followed by a colon, and separator omission is permitted, 
-    // expect that we'll parse another list element which begins with that 
-    // argument label and continue parsing the list.
-    if (Tok.isAtStartOfLine() && AllowSepOmission && Tok.canBeArgumentLabel() 
-        && peekToken().is(tok::colon)) { 
-      continue;
-    }
-
-    // If we're in a comma-separated list, the next token is at the
-    // beginning of a new line and can never start an element, break.
-    if (Tok.isAtStartOfLine() &&
-        (Tok.is(tok::r_brace) || isStartOfDecl() || isStartOfStmt())) {
-      break;
+    if (Tok.isAtStartOfLine()) { 
+      if (AllowSepOmission && Tok.canBeArgumentLabel() 
+          && peekToken().is(tok::colon)) 
```

Comment:
One more nit I glanced over: we use `clang-format` which makes this code incorrectly formatted. The workflow I have to add run git-clang-format on the files I have ready to be committed to make sure everything is correctly formatted. The tool is shipped together with our swift-clang so you just need to add `<swift-dir>/clang/tools/clang-format` to your `PATH` to get that working.

Reply:
Thanks!  I'll set my system up that way.  Fixed the whitespace in https://github.com/apple/swift/pull/22714/commits/2adc41d697485f395c14294a2f1b8e8b796e32e1 .